cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -29,92 +29,141 @@ import (
 type MasterConfiguration struct {
 	metav1.TypeMeta
 
-	API                  API
-	KubeProxy            KubeProxy
-	Etcd                 Etcd
+	// API holds configuration for the k8s api server.
+	API API
+	// KubeProxy holds configuration for the k8s service proxy.
+	KubeProxy KubeProxy
+	// Etcd holds configuration for etcd.
+	Etcd Etcd
+	// KubeletConfiguration holds configuration for kubelet.
 	KubeletConfiguration KubeletConfiguration
-	Networking           Networking
-	KubernetesVersion    string
-	CloudProvider        string
-	NodeName             string
-	AuthorizationModes   []string
+	// Networking holds configuration for the networking topology of the cluster.
+	Networking Networking
+	// KubernetesVersion is the target version of the control plane.
+	KubernetesVersion string
+	// CloudProvider is the name of the cloud provider.
+	CloudProvider string
+	// NodeName is the name of the node that will host the k8s control plane.
+	// Defaults to the name of the host if not provided.
+	NodeName string
+	// AuthorizationModes is a set of authorization modes used inside the cluster.
+	// If not specified, defaults to Node and RBAC, meaning both the node
+	// authorizer and RBAC are enabled.
+	AuthorizationModes []string
 
 	// Mark the controller and api server pods as privileged as some cloud
 	// controllers like openstack need escalated privileges under some conditions
-	// example - loading a config drive to fetch node information
+	// example - loading a config drive to fetch node information.
 	PrivilegedPods bool
 
-	Token    string
+	// Token is used for establishing bidirectional trust between nodes and masters.
+	// Used for joining nodes in the cluster.
+	Token string
+	// TokenTTL is a ttl for Token. Defaults to 24h.
 	TokenTTL *metav1.Duration
 
-	APIServerExtraArgs         map[string]string
+	// APIServerExtraArgs is a set of extra flags to pass to the API Server or override
+	// default ones in form of <flagname>=<value>.
```

Comment:
Could you add a // TODO: This is temporary and ideally we would like to switch all components to use ComponentConfig + ConfigMaps.